internal/ethapi/api.go

```diff
@@ -897,6 +897,43 @@ func (s *BlockChainAPI) GetStorageAt(ctx context.Context, address common.Address
 	return res[:], state.Error()
 }
 
+// GetBlockReceipts returns the block receipts for the given block hash or number or tag.
+func (s *BlockChainAPI) GetBlockReceipts(ctx context.Context, blockNrOrHash rpc.BlockNumberOrHash) ([]map[string]interface{}, error) {
+	var hash common.Hash
+	if h, ok := blockNrOrHash.Hash(); ok {
+		hash = h
+	} else {
+		block, err := s.b.BlockByNumberOrHash(ctx, blockNrOrHash)
+		if err != nil {
+			return nil, err
+		}
+		hash = block.Hash()
+	}
+
+	block, err := s.b.BlockByHash(ctx, hash)
```

Comment:
Why not directly use `s.b.BlockByNumberOrHash`? IMO not worth the extra 10 lines to save on hashing a block.

Reply:
thanks for this 👍